File: POO/tad Lista/C++/list.cpp
#include <iostream>
#include "list.h"

List::List() : size(0), vector(nullptr) {}

List::~List()
{
    delete[] vector;
}

void List::insert(int value)
{
    size++;
    int *newVector = new int[size];
    for (int x = 0; x < size - 1; x++)
    {
        newVector[x] = vector[x];
    }
    newVector[size - 1] = value;
    delete[] vector;
    vector = newVector;
}

void List::removeLast()
{
    if (size == 0)
    {
        std::cout << "Lista Vazia!" << std::endl;
        return;
    }
    size--;
    int *newVector = new int[size];
    for (int x = 0; x < size; x++)
    {
        newVector[x] = vector[x];
    }
    delete[] vector;
    vector = newVector;
}

int List::firstItem()
{
    if (size == 0)
    {
        std::cout << "Lista Vazia!" << std::endl;
        return -1;
    }
    return vector[0];
}

int List::lastItem()
{
    if (size == 0)
    {
        std::cout << "Lista Vazia!" << std::endl;
        return -1;
    }
    return vector[size - 1];
}

int List::getSize()
{
    return size;
}

File: POO/tad Lista/java/List.java
public class List 
{
    private int size;
    private int[] vector;

    public List() 
    {
        this.size = 0;
        this.vector = new int[0];
    }

    public void insert(int value) 
    {
        int[] newVector = new int[size + 1];
        for (int x = 0; x < size; x++) 
        {
            newVector[x] = vector[x];
        }
        newVector[size] = value;
        vector = newVector;
        size++;
    }

    public void removeFirst() 
    {
        if (size > 0) 
        {
            if (size == 1) 
            {
                vector = new int[0];
            } else {
                int[] newVector = new int[size - 1];
                for (int x = 0; x < size - 1; x++) 
                {
                    newVector[x] = vector[x + 1];
                }
                vector = newVector;
            }
            size--;
        } else 
        {
            System.out.println("Lista Vazia!");
        }
    }
    
    public void removeLast() 
    {
        if (size > 0) 
        {
            if (size == 1) 
            {
                vector = new int[0];
            } else {
                int[] newVector = new int[size - 1];
                for (int x = 0; x < size - 1; x++) 
                {
                    newVector[x] = vector[x];
                }
                vector = newVector;
            }
            size--;
        } else 
        {
            System.out.println("Lista Vazia!");
        }
    }

    public int firstItem() 
    {
        if (size == 0) 
        {
            System.out.println("Lista Vazia!");
            return -1;
        } else 
        {
            return vector[0];
        }
    }

    public int lastItem() 
    {
        if (size == 0) 
        {
            System.out.println("Lista Vazia!");
            return -1;
        } else 
        {
            return vector[size - 1];
        }
    }

    public int getSize() 
    {
        return size;
    }
}
